lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -123,9 +125,9 @@ def self.expose(*args)
           return options[:override_return] if options.key?(:override_return)
           reflection = @object.class.reflection_with_virtual(method)
           if reflection && reflection.collection?
-            wrap_results(ret.to_a)
+            options[:association] ? wrap_results(self.class.filter_objects(ret.to_a)) : wrap_results(ret.to_a)
```

Comment:
can you remind me why you need the `ret.to_a`?
Have you tried removing the `to_a`?
Besides a very rare edge case, it should work and speed things up considerably.

(I've mentioned this repeatedly in the last pr)